Auth0 Migration instructions (DO NOT MERGE UNTIL AUTH0 MIGRATION IS LIVE)

matthewshaver opened the pull request:
## What are you changing in this pull request and why?
Migration instructions for Auth0 SSO
https://dbtlabs.atlassian.net/browse/ENTERPRISE-529

[Preview](https://deploy-preview-3158--docs-getdbt-com.netlify.app/docs/cloud/manage-access/auth0-migration) 

## Checklist

- [x] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) and [About versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) so my content adheres to these guidelines.
- [x] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

Adding new pages (delete if not applicable):
- [x] Add page to `website/sidebars.js`
- [x] Provide a unique filename for the new page

matthewshaver committed:
Auth0 Migration instructions

matthewshaver committed:
updates to migration guide

matthewshaver committed:
Updates to SSO URIs

ajaythomas commented on website/docs/docs/cloud/manage-access/auth0-migration.md:
```diff
@@ -0,0 +1,60 @@
+
+The fields that will be updated are:
+- Single sign-on URL &mdash; `https://us-devspace-cluster.us.auth0.com/login/callback?connection={slug}`
+- Audience URI (SP Entity ID) &mdash; `urn:auth0:us-devspace-cluster:{slug}`
```
@matthewshaver We could use the pattern we use today. Using a placeholder (The following steps use YOUR_ACCESS_URL which needs to be replaced with the [appropriate Access URL](https://docs.getdbt.com/docs/cloud/about-cloud/regions-ip-addresses) for your region and plan.) that points to a table for the different Auth0 tenants instead of hard-coding `us-devspace-cluster.us.auth0.com`. Custom domains for prod is coming which will make these urls look like [this](https://www.notion.so/dbtlabs/Auth0-tenants-and-custom-domains-7e8c90aee9d541e8b69bc5f3eb3e6206?pvs=4#af17f595f2aa44e2906b810d3f275b61)

https://docs.getdbt.com/docs/cloud/manage-access/set-up-sso-saml-2.0

cc @robotmiller @thorn14 @saimaddali

aaronbini replied:
agree with Ajay on this that it would be best to follow the existing pattern. [this PR](https://github.com/dbt-labs/docs.getdbt.com/pull/2933) I opened a while ago updates the SSO setup docs on the `next` version of the docs site, and something similar could be followed here I think.

matthewshaver committed:
Adding Azure content

carusocv commented on website/docs/docs/cloud/manage-access/auth0-migration.md:
```diff
@@ -0,0 +1,80 @@
+description: "Required actions for migrating to Auth0 for SSO services on dbt Cloud"
+---
+
+dbt Labs is partnering with Auth0 to bring enhanced features to dbt Cloud's single sign-on (SSO) capabilities. Auth0 is an identity and access management (IAM) platform with advanced security features, and it will be the dbt Cloud SSO identity provider (IDP) moving forward. These changes will require some action from customers with SSO configured in dbt Cloud today, and this guide will outline th ...
```
> and it will be the dbt Cloud SSO identity provider (IDP) moving forward.

Technically I'm not sure if this is true (at least for the current rollout*). My understanding is that while we are using the Auth0 engine/integration - the SSO providers (Okta, AD etc) are still technically the Identity Provider. Similarly, for password logins, dbt Cloud is the source of truth and therefor we are the IdP for password logins. 

From Auth0 website

> Auth0 provides IdP-initiated SSO Integrations for various services

So I'm not sure its the best way to communicate this. Maybe to 99% of users it is the same thing - but want to be extra clear here for the 1%

*Note: eventually we will use the Auth0 user store for password logins in which case Auth0 will be the Identity Provider for password logins.

matthewshaver replied:
Thank you! Altered the text to provide clarity.

carusocv commented on website/docs/docs/cloud/manage-access/auth0-migration.md:
```diff
@@ -0,0 +1,80 @@
+
+Once you have saved this information in the SSO environment, you must edit some information in the dbt Cloud settings. Navigate to the **Account Settings**, update the single sign-on URL fields, and provide the updated x.509 certificate.
+
+Make sure to enable the "Use Auth0" option to ensure that the traffic is routed correctly. _The Auth0 migration action is final and cannot be undone_
```
Just want to check/confirm what the plan is if a customer accidentally toggle this. Both for communication (i.e they need to reach out support), and for the playbook on our end. Hopefully this doesn't happen, but odds are high enough it may occur once or twice.

BrJan replied:
@robotmiller worked on this if we need to roll back an account - https://dbtlabs.atlassian.net/browse/ENTERPRISE-667.

matthewshaver committed:
Updating URIs and adding snippet

matthewshaver committed:
Minor corrections

matthewshaver committed:
Apply suggestions from code review

matthewshaver committed:
Update set-up-sso-google-workspace.md

matthewshaver committed:
Update set-up-sso-okta.md

matthewshaver committed:
Update set-up-sso-saml-2.0.md

matthewshaver committed:
Merge branch 'current' into update-sso

matthewshaver committed:
Merge branch 'current' into auth0-migration

matthewshaver committed:
Adding Azure MT

matthewshaver committed:
Merge branch 'update-sso' of https://github.com/dbt-labs/docs.getdbt.com into update-sso

ajaythomas commented on website/static/img/docs/dbt-cloud/access-control/enable-auth0.png:
Can we use screenshots from US MT instead of devspace cluster.

ajaythomas commented on website/static/img/docs/dbt-cloud/access-control/google-uri.png:
Can we use screenshots from US MT instead of devspace cluster.

ajaythomas commented on website/static/img/docs/dbt-cloud/access-control/new-okta-completed.png:
Can we use screenshots from US MT instead of devspace cluster.

ajaythomas commented on website/docs/docs/cloud/manage-access/auth0-migration.md:
```diff
@@ -0,0 +1,82 @@
+
+4. _You will need a person with Google Workspace admin privileges to complete these steps in dbt Cloud_. In dbt Cloud, navigate to the **Account Settings**, click on **Single Sign-on**, and then click **Edit** on the right side of the SSO pane. Enable the **Use Auth0** option and select **Save**. This will trigger an authorization window from Google that will require admin credentials. _The Auth0 ...
+
+<Lightbox src="/img/docs/dbt-cloud/access-control/enable-auth0.png" title="Enable Auth0"/>
```
This screenshot `website/static/img/docs/dbt-cloud/access-control/enable-auth0.png` is for SAML not google.

matthewshaver replied:
Fixed!

ajaythomas commented on website/docs/docs/cloud/manage-access/auth0-migration.md:
```diff
@@ -0,0 +1,82 @@
+
+4. Navigate to the dbt Cloud environment and open the **Account Settings**. Click the **Single Sign-on** option from the left side menu and click the **Edit** option from the right side of the SSO pane. Select the **Enable Auth0** option and **Save**. _Once this option is enabled, it cannot be undone._
+
+<Lightbox src="/img/docs/dbt-cloud/access-control/enable-auth0.png" title="Enable Auth0"/>
```
This screenshot `website/static/img/docs/dbt-cloud/access-control/enable-auth0.png` is for SAML not Azure.

matthewshaver replied:
Resolved by narrowing the screenshot

BrJan commented:
@matthewshaver about this -

"You can approach this by editing the existing SSO app settings or creating a new one to accommodate the Auth0 settings. One approach isn't inherently better, so choose whichever works best for your organization."

matthewshaver committed:
Updates to SSO URIs (DO NOT PUBLISH UNTIL AUTH0 IS LIVE) (#3166)

## What are you changing in this pull request and why?
Updating the URIs for the new Auth0 workflows with placeholders. The
final URIs will be subbed in when available

## Checklist
<!--
Uncomment if you're publishing docs for a prerelease version of dbt
(delete if not applicable):
- [ ] Add versioning components, as described in [Versioning
Docs](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-entire-pages)
- [ ] Add a note to the prerelease version [Migration
Guide](https://github.com/dbt-labs/docs.getdbt.com/tree/current/website/docs/guides/migration/versions)
-->
- [ ] Review the [Content style
guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md)
and [About
versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version)
so my content adheres to these guidelines.
- [ ] Add a checklist item for anything that needs to happen before this
PR is merged, such as "needs technical review" or "change base branch."

Adding new pages (delete if not applicable):
- [ ] Add page to `website/sidebars.js`
- [ ] Provide a unique filename for the new page

Removing or renaming existing pages (delete if not applicable):
- [ ] Remove page from `website/sidebars.js`
- [ ] Add an entry `_redirects`
- [ ] [Ran link
testing](https://github.com/dbt-labs/docs.getdbt.com#running-the-cypress-tests-locally)
to update the links that point to the deleted page

matthewshaver committed:
Merge branch 'current' into auth0-migration

matthewshaver committed:
Updating screenshot

matthewshaver committed:
Delete set-up-sso-azure-active-directory.md

matthewshaver committed:
Fixing a mistake

matthewshaver commented on website/docs/docs/cloud/manage-access/set-up-sso-google-workspace.md:
```diff
@@ -65,7 +67,7 @@ Client Secret for use in dbt Cloud.
 | **Name** | dbt Cloud |
 | **Authorized Javascript origins** | `https://YOUR_ACCESS_URL` |
-| **Authorized Redirect URIs** | `https://YOUR_ACCESS_URL/complete/gsuite` |
+| **Authorized Redirect URIs** | `https://<YOUR_AUTH0_URI>/login/callback` |
```
```suggestion
| **Authorized Redirect URIs** | `https://YOUR_ACCESS_URL/complete/gsuite` |
```

matthewshaver commented on website/docs/docs/cloud/manage-access/set-up-sso-google-workspace.md:
```diff
@@ -29,6 +29,8 @@ data workloads on AWS.
 This guide outlines the setup process for authenticating to dbt Cloud with
 Google GSuite.
 
+<Snippet src="auth0-uri" />
```
```suggestion
```

matthewshaver commented on website/docs/docs/cloud/manage-access/set-up-sso-okta.md:
```diff
@@ -21,6 +21,8 @@ This guide outlines the setup process for authenticating to dbt Cloud with Okta.
 If you have any questions during the setup process, please contact support
 (support@getdbt.com) for assistance.
 
+<Snippet src="auth0-uri" />
```
```suggestion
```

matthewshaver commented on website/docs/docs/cloud/manage-access/set-up-sso-okta.md:
```diff
@@ -76,13 +78,13 @@ so pick a slug that uniquely identifies your company.
 
-* **Single sign on URL**: `https://YOUR_ACCESS_URL/complete/okta`
-* **Audience URI (SP Entity ID)**: `https://YOUR_ACCESS_URL/`
+* **Single sign on URL**: `https://<YOUR_AUTH0_URI>/login/callback?connection={slug}`
```
```suggestion
* **Single sign on URL**: `https://YOUR_ACCESS_URL/complete/okta`
```

matthewshaver commented on website/docs/docs/cloud/manage-access/set-up-sso-okta.md:
```diff
@@ -76,13 +78,13 @@ so pick a slug that uniquely identifies your company.
-* **Single sign on URL**: `https://YOUR_ACCESS_URL/complete/okta`
-* **Audience URI (SP Entity ID)**: `https://YOUR_ACCESS_URL/`
+* **Single sign on URL**: `https://<YOUR_AUTH0_URI>/login/callback?connection={slug}`
+* **Audience URI (SP Entity ID)**: `urn:<YOUR_AUTH0_ENTITYID>:{slug}`
```
```suggestion
* **Audience URI (SP Entity ID)**: `https://YOUR_ACCESS_URL/`
```

matthewshaver commented on website/docs/docs/cloud/manage-access/set-up-sso-okta.md:
```diff
@@ -76,13 +78,13 @@ so pick a slug that uniquely identifies your company.
 <Lightbox
     collapsed={false}
-    src="/img/docs/dbt-cloud/dbt-cloud-enterprise/okta/okta-3-saml-settings-top.png"
+    src="/img/docs/dbt-cloud/access-control/new-okta-config.png"
```
```suggestion
    src="/img/docs/dbt-cloud/dbt-cloud-enterprise/okta/okta-3-saml-settings-top.png"
```

matthewshaver commented on website/docs/docs/cloud/manage-access/set-up-sso-saml-2.0.md:
```diff
@@ -19,6 +19,8 @@ Currently supported features include:
 This document details the steps to integrate dbt Cloud with an identity
 provider in order to configure Single Sign On and [role-based access control](/docs/cloud/manage-access/about-user-access#role-based-access-control).
 
+<Snippet src="auth0-uri" />
```
```suggestion
```

matthewshaver commented on website/docs/docs/cloud/manage-access/set-up-sso-saml-2.0.md:
```diff
@@ -46,8 +48,8 @@ Login slugs must be unique across all dbt Cloud accounts, so pick a slug that un
 
-* Single sign on URL: `https://YOUR_ACCESS_URL/complete/saml`
-* Audience URI (SP Entity ID): `https://YOUR_ACCESS_URL/`
+- Single sign on URL: `https://<YOUR_AUTH0_URI>/login/callback?connection={slug}`
```
```suggestion
* Single sign on URL: `https://YOUR_ACCESS_URL/complete/saml`
```

matthewshaver commented on website/docs/docs/cloud/manage-access/set-up-sso-saml-2.0.md:
```diff
@@ -46,8 +48,8 @@ Login slugs must be unique across all dbt Cloud accounts, so pick a slug that un
-* Single sign on URL: `https://YOUR_ACCESS_URL/complete/saml`
-* Audience URI (SP Entity ID): `https://YOUR_ACCESS_URL/`
+- Single sign on URL: `https://<YOUR_AUTH0_URI>/login/callback?connection={slug}`
+- Audience URI (SP Entity ID): `urn:auth0:us-devspace-cluster:{slug}`
```
```suggestion
* Audience URI (SP Entity ID): `https://YOUR_ACCESS_URL/`
```

matthewshaver commented on website/docs/docs/cloud/manage-access/set-up-sso-saml-2.0.md:
```diff
@@ -140,9 +142,9 @@ Login slugs must be unique across all dbt Cloud accounts, so pick a slug that un
-   * **Single sign on URL**: `https://YOUR_ACCESS_URL/complete/okta`
-   * **Audience URI (SP Entity ID)**: `https://YOUR_ACCESS_URL/`
-   * **Relay State**: `<login slug>`
+   - Single sign on URL: `https://<YOUR_AUTH0_URI>/login/callback?connection={slug}`
```
```suggestion
   * **Single sign on URL**: `https://YOUR_ACCESS_URL/complete/okta`
```

matthewshaver commented on website/docs/docs/cloud/manage-access/set-up-sso-saml-2.0.md:
```diff
@@ -140,9 +142,9 @@ Login slugs must be unique across all dbt Cloud accounts, so pick a slug that un
-   * **Audience URI (SP Entity ID)**: `https://YOUR_ACCESS_URL/`
-   * **Relay State**: `<login slug>`
+   - Single sign on URL: `https://<YOUR_AUTH0_URI>/login/callback?connection={slug}`
+   - Audience URI (SP Entity ID): `urn:<YOUR_AUTH0_ENTITYID>:{slug}`
```
```suggestion
      * **Audience URI (SP Entity ID)**: `https://YOUR_ACCESS_URL/`

matthewshaver commented on website/docs/docs/cloud/manage-access/set-up-sso-saml-2.0.md:
```diff
@@ -140,9 +142,9 @@ Login slugs must be unique across all dbt Cloud accounts, so pick a slug that un
-   * **Relay State**: `<login slug>`
+   - Single sign on URL: `https://<YOUR_AUTH0_URI>/login/callback?connection={slug}`
+   - Audience URI (SP Entity ID): `urn:<YOUR_AUTH0_ENTITYID>:{slug}`
+   - Relay State: `<login slug>`
```
```suggestion
   * **Relay State**: `<login slug>`
```

matthewshaver replied:
```suggestion
   * **Relay State**: `<login slug>`
```

matthewshaver committed:
Apply suggestions from code review

matthewshaver commented on website/docs/docs/cloud/manage-access/set-up-sso-saml-2.0.md:
```diff
@@ -241,9 +242,9 @@ Login slugs must be unique across all dbt Cloud accounts, so pick a slug that un
-   * **ACS URL**: `https://YOUR_ACCESS_URL/complete/saml`
-   * **Audience URI (SP Entity ID)**: `https://YOUR_ACCESS_URL/`
-   - **Start URL**: `<login slug>`
+   - Single sign on URL: `https://<YOUR_AUTH0_URI>/login/callback?connection={slug}`
```
```suggestion
   * **ACS URL**: `https://YOUR_ACCESS_URL/complete/saml`
```

matthewshaver commented on website/docs/docs/cloud/manage-access/set-up-sso-saml-2.0.md:
```diff
@@ -241,9 +242,9 @@ Login slugs must be unique across all dbt Cloud accounts, so pick a slug that un
-   * **Audience URI (SP Entity ID)**: `https://YOUR_ACCESS_URL/`
-   - **Start URL**: `<login slug>`
+   - Single sign on URL: `https://<YOUR_AUTH0_URI>/login/callback?connection={slug}`
+   - Audience URI (SP Entity ID): `urn:<YOUR_AUTH0_ENTITYID>:{slug}`
```
```suggestion
   * **Audience URI (SP Entity ID)**: `https://YOUR_ACCESS_URL/`
```

matthewshaver commented on website/docs/docs/cloud/manage-access/set-up-sso-saml-2.0.md:
```diff
@@ -241,9 +242,9 @@ Login slugs must be unique across all dbt Cloud accounts, so pick a slug that un
-   - **Start URL**: `<login slug>`
+   - Single sign on URL: `https://<YOUR_AUTH0_URI>/login/callback?connection={slug}`
+   - Audience URI (SP Entity ID): `urn:<YOUR_AUTH0_ENTITYID>:{slug}`
+   - Relay State: `<login slug>`
```
```suggestion
   - **Start URL**: `<login slug>`
```

matthewshaver committed:
Apply suggestions from code review

matthewshaver commented on website/docs/docs/cloud/manage-access/set-up-sso-saml-2.0.md:
```diff
@@ -312,7 +313,7 @@ Follow these steps to set up single sign-on (SSO) with dbt Cloud:
    | ----- | ----- |
    | **Identifier (Entity ID)** | Base URL for dbt Cloud. Use `https://YOUR_ACCESS_URL/`. |
-   | **Reply URL (Assertion Consumer Service URL)** | Use `https://YOUR_ACCESS_URL/complete/saml`. |
+   | **Reply URL (Assertion Consumer Service URL)** | Use `https://<YOUR_AUTH0_URI>/login/callback?connection={slug}`. |
```
```suggestion
   | **Reply URL (Assertion Consumer Service URL)** | Use `https://YOUR_ACCESS_URL/complete/saml`. |
```

matthewshaver commented on website/docs/docs/cloud/manage-access/set-up-sso-saml-2.0.md:
```diff
@@ -369,9 +370,9 @@ Login slugs must be unique across all dbt Cloud accounts, so pick a slug that un
-   - **Audience (EntityID):** https://YOUR_ACCESS_URL/
-   - **ACS (Consumer) URL Validator:** https://YOUR_ACCESS_URL/complete/saml
-   - **ACS (Consumer) URL:** https://YOUR_ACCESS_URL/complete/saml
+   - **Audience (EntityID):** `urn:<YOUR_AUTH0_ENTITYID>:{slug}`
```
```suggestion
   - **Audience (EntityID):** https://YOUR_ACCESS_URL/
```

matthewshaver commented on website/docs/docs/cloud/manage-access/set-up-sso-saml-2.0.md:
```diff
@@ -369,9 +370,9 @@ Login slugs must be unique across all dbt Cloud accounts, so pick a slug that un
-   - **ACS (Consumer) URL Validator:** https://YOUR_ACCESS_URL/complete/saml
-   - **ACS (Consumer) URL:** https://YOUR_ACCESS_URL/complete/saml
+   - **Audience (EntityID):** `urn:<YOUR_AUTH0_ENTITYID>:{slug}`
+   - **ACS (Consumer) URL Validator:** https://<YOUR_AUTH0_URI>/login/callback?connection={slug}
```
```suggestion
   - **ACS (Consumer) URL Validator:** https://YOUR_ACCESS_URL/complete/saml
```

matthewshaver commented on website/docs/docs/cloud/manage-access/set-up-sso-saml-2.0.md:
```diff
@@ -369,9 +370,9 @@ Login slugs must be unique across all dbt Cloud accounts, so pick a slug that un
-   - **ACS (Consumer) URL:** https://YOUR_ACCESS_URL/complete/saml
+   - **Audience (EntityID):** `urn:<YOUR_AUTH0_ENTITYID>:{slug}`
+   - **ACS (Consumer) URL Validator:** https://<YOUR_AUTH0_URI>/login/callback?connection={slug}
+   - **ACS (Consumer) URL:** https://<YOUR_AUTH0_URI>/login/callback?connection={slug}
```
```suggestion
   - **ACS (Consumer) URL:** https://YOUR_ACCESS_URL/complete/saml
```

matthewshaver committed:
Apply suggestions from code review

matthewshaver committed:
updating screenshots

matthewshaver committed:
Merge branch 'auth0-migration' of https://github.com/dbt-labs/docs.getdbt.com into auth0-migration

matthewshaver committed:
Adding screenshot

matthewshaver commented on website/docs/docs/cloud/manage-access/set-up-sso-saml-2.0.md:
```diff
@@ -141,7 +142,7 @@ Login slugs must be unique across all dbt Cloud accounts, so pick a slug that un
 
    * **Single sign on URL**: `https://YOUR_ACCESS_URL/complete/okta`
-   * **Audience URI (SP Entity ID)**: `https://YOUR_ACCESS_URL/`
+      * **Audience URI (SP Entity ID)**: `https://YOUR_ACCESS_URL/`
```
```suggestion
   * **Audience URI (SP Entity ID)**: `https://YOUR_ACCESS_URL/`
```

matthewshaver committed:
Update website/docs/docs/cloud/manage-access/set-up-sso-saml-2.0.md

matthewshaver commented on website/docs/docs/cloud/manage-access/set-up-sso-saml-2.0.md:
```diff
@@ -19,6 +19,7 @@ Currently supported features include:
 This document details the steps to integrate dbt Cloud with an identity
 provider in order to configure Single Sign On and [role-based access control](/docs/cloud/manage-access/about-user-access#role-based-access-control).
 
+
```
```suggestion
```

matthewshaver committed:
Apply suggestions from code review

matthewshaver commented on website/docs/docs/cloud/manage-access/set-up-sso-google-workspace.md:
```diff
@@ -29,6 +29,7 @@ data workloads on AWS.
 This guide outlines the setup process for authenticating to dbt Cloud with
 Google GSuite.
 
+
```
```suggestion
```

matthewshaver commented on website/docs/docs/cloud/manage-access/set-up-sso-okta.md:
```diff
@@ -21,6 +21,7 @@ This guide outlines the setup process for authenticating to dbt Cloud with Okta.
 If you have any questions during the setup process, please contact support
 (support@getdbt.com) for assistance.
 
+
```
```suggestion
```

matthewshaver committed:
Apply suggestions from code review

runleonarun reviewed (approved):
:shipit:

ajaythomas commented:
@matthewshaver @BrJan for `website/static/img/docs/dbt-cloud/access-control/new-okta-completed.png` file in this PR, can we use a screenshot that has custom domain unlike devspace-cluster.
This screenshot is a good one from au.

https://www.notion.so/dbtlabs/WIP-Auth0-Corbato-Overhaul-Support-Readiness-0d388b0ed9df4819a8dc997e083b923b?pvs=4#5671fd1f1ecd46a3bd036473cb58ad2c
